src/HikConsole/HikClient.cs

```diff
@@ -1,181 +1,185 @@
+                this.hikApi.StopDownloadFile(this.downloadId);
                 this.ResetDownloadStatus();
             }
+            else
+            {
+                this.logger.Warn("StopDownload() : File not downloading now");
+            }
         }
 
-        public void CheckProgress()
+        public void UpdateProgress()
         {
             if (this.IsDownloading)
             {
-                int barValue = this.sdk.GetDownloadPos(this.downloadHandle);
-
-                if (barValue > ProgressBarMinimum && barValue < ProgressBarMaximum)
-                {
-                    this.progress?.Report((double)barValue / 100);
-                }
-                else if (barValue == 100)
-                {
-                    this.StopDownload();
-                    this.downloadFile = null;
-
-                    C.WriteLine("- downloaded", ConsoleColor.Green);
-                }
-                else if (barValue == 200)
-                {
-                    C.WriteLine("The downloading is abnormal for the abnormal network!", ConsoleColor.DarkRed);
+                int downloadProgress = this.hikApi.GetDownloadPosition(this.downloadId);
 
-                    this.ForceExit();
-                }
+                this.UpdateDownloadProgress(downloadProgress);
+            }
+            else
+            {
+                this.logger.Warn(" UpdateProgress() : File not downloading now");
             }
         }
 
         public void Logout()
         {
-            if (this.userId >= 0)
+            if (this.loginResult != null)
+            {
+                this.logger.Info($"Logout the device");
+                this.hikApi.Logout(this.loginResult.UserId);
+                this.hikApi.Cleanup();
+                this.loginResult = null;
+            }
+            else
             {
-                C.WriteLine($"Logout the device", timeStamp: DateTime.Now);
-                this.sdk.Logout(this.userId);
-                this.sdk.Cleanup();
-                this.userId = -1;
+                this.logger.Warn("Logout() : Session ended");
             }
         }
 
         public void ForceExit()
         {
-            C.WriteLine("\r\nForce exit", ConsoleColor.DarkRed);
+            this.logger.Warn("\r\nForce exit");
             this.StopDownload();
             this.DeleteCurrentFile();
             this.Logout();
         }
 
-        public async Task<IList<FindResult>> Find(DateTime periodStart, DateTime periodEnd)
+        public Task<IList<RemoteVideoFile>> FindAsync(DateTime periodStart, DateTime periodEnd)
         {
             this.ValidateDateParameters(periodStart, periodEnd);
 
-            return await this.Find(periodStart, periodEnd, this.userId, this.channel);
+            return this.SearchVideoFilesAsync(periodStart, periodEnd, this.loginResult);
         }
 
-        private void PrintFileInfo(FindResult file)
+        private string GetPrintableFileInfo(RemoteVideoFile file)
```

Comment:
You can define Formarters for RemoteVideoFile  and use it for printing data or serializing